### PR TITLE
New pytests for sparkl call

### PR DESCRIPTION
**Pull request by jacobythwaites**
Tests to check call functionality works with literal and file variable values.

### CODE REVIEW

**Comment by miklosduma** [comment]
SPARKL login details (username, password) should not be disclosed. Using a token would be better.

File: Makefile [hunk 1]
```diff
@@ -2,6 +2,13 @@
 # Author <jacoby@sparkl.com> Jacoby Thwaites
 VERSION := $(shell git describe --tags --long --abbrev=1)
 PY_VERSION := $(shell git describe --tags --abbrev=0)
+
+########################
+# DEFAULT TARGET IS HELP
+########################
```
Comment (reube) [inline Makefile]: :+1:

File: Makefile [hunk 2]
```diff
@@ -39,9 +46,33 @@ doc:
...
+endif
+
+.PHONY: TEST_PASS
+ifdef TEST_PASS
+TEST_PASS: ;
+else
+TEST_PASS:
+	$(error Must set env var: $@)
+endif
+
```
Comment (reube) [inline Makefile]: Ack

File: sparkl_cli/cmd_call.py [hunk 3]
```diff
@@ -52,9 +58,51 @@ def sparkl_type(field_type, string_value):
...
+    TYPE             LITERAL EXAMPLE.      READ FROM FILE
+    -------------    -------------------   -------------------------------
+    boolean          String, "true"        File read as string
+    float            String, "3.14"        File read as string
+    integer          String, "314"         File read as string
+    utf8             String, "Hello"       File read as string
+    json             JSON,   "\"hi\""      File read as string
+    binary           Base64, "SGk="        Binary file converted to base64
+    erlang string    String, "Hello"       File read as string
+    erlang term      String, "{a,b}"       File read as string
```
Comment (reube) [inline sparkl_cli/cmd_call.py]: Good commentary.

**Pull request merged**